### ISSUE TITLE
`Module` bug

### ISSUE THREAD

**Issue by ASEM000** [patrick-kidger/equinox#481]
Hello patrick, 

I think the following is a bug in `Module`

```python
import equinox as eqx

class ParentTree(eqx.Module):
    def __init_subclass__(klass: type, foo: bool = False):
        if foo:
            klass.foo = True

class Tree(ParentTree, foo=True):
    def __init__(self, a: int):
        self.a = a

Tree(a=1).foo
# _ModuleMeta.__new__() got an unexpected keyword argument 'foo'
```
if i omit `foo` a different error comes up:

```python
class Tree(ParentTree):
    def __init__(self, a: int):
        self.a = a

Tree(a=1)
# AttributeError: Cannot set attribute a
```

In comparison to [pytreeclass](https://github.com/ASEM000/pytreeclass) or [simple_pytree](https://github.com/cgarciae/simple-pytree), this works as expected:

```python
import pytreeclass as pytc
class ParentTree(pytc.TreeClass):
    def __init_subclass__(klass: type, foo: bool = False):
        if foo:
            klass.foo = True
class Tree(ParentTree, foo=True):
    def __init__(self, a: int):
        self.a = a
Tree(a=1).foo
# True
```

```python
from simple_pytree import Pytree
class ParentTree(Pytree):
    def __init_subclass__(klass: type, foo: bool = False):
        if foo:
            klass.foo = True
class Tree(ParentTree, foo=True):
    def __init__(self, a: int):
        self.a = a
Tree(a=1).foo
# True
```

The`Module` logic became more complex since the last time i checked, so Im not sure the reason of the two errors. But one pointer is that _maybe_ propagate kwargs in module meta new.

**Comment by patrick-kidger** [comment]
Regarding `foo=True`: thanks! This'll be fixed by #482.

Regarding omitting `foo`: Equinox's syntax involves explicitly declaring all attributes. You want
```python
class Tree(ParentTree):
    a: int  # this bit is new

    def __init__(self, a: int):
        self.a = a
```
although also note that the default `__init__` method matches the one explicity provided, so that can be skipped. Both the attribute declaration and the default `__init__` are the same behaviour as with any dataclass.

**Comment by ASEM000** [comment]
This was quick :D

**Comment by ASEM000** [comment]
> Regarding omitting foo: Equinox's syntax involves explicitly declaring all attributes.

My bad.

**Comment by patrick-kidger** [comment]
Closing! Fixed in #482, and this will appear in the upcoming v0.11.0 release.
